Added a new optional dark theme that provides greater contrast for the text in the editor on all supported platforms (#41234)
mantidproject/mantid: docs/source/workbench/settings.rst

Reviewer: Sorry for asking this, just to be a bit precise with the instructions, could we embed the editor settings in a groupbox?

<img width="400" height="225" alt="Image" src="https://github.com/user-attachments/assets/9ca69687-8789-409c-a77a-5059345cf6ea" />

- Confirmations and Notifications: These settings enable and disable confirmation boxes to save scripts
  and projects and error notifications.
- Project Recovery: These settings enable project recovery in case of a crash (recommended).

> - Editor: These settings controls the option of enabling auto-completion or dark theme in the script editor.

Reviewer: ```suggestion
- Editor: These settings toggle the auto-completion and dark theme display in the script editor.
```
